ibis/expr/types/temporal.py

```diff
@@ -474,6 +474,43 @@ def delta(
         """
         return ops.DateDelta(left=self, right=other, part=part).to_expr()
 
+    def epoch_days(self) -> ir.IntegerValue:
+        """Return the number of days since the UNIX epoch date.
+
+        Examples
+        --------
+        >>> from datetime import date
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> d = date(2020, 1, 1)
+        >>> ibis.date(d)
+        ┌────────────┐
+        │ 2020-01-01 │
+        └────────────┘
+        >>> ibis.date(d).epoch_days()
+        ┌───────┐
+        │ 18262 │
+        └───────┘
+        >>> t = ibis.memtable({"date_col": [d]})
+        >>> t
+        ┏━━━━━━━━━━━━┓
+        ┃ date_col   ┃
+        ┡━━━━━━━━━━━━┩
+        │ date       │
+        ├────────────┤
+        │ 2020-01-01 │
+        └────────────┘
+        >>> t.mutate(epoch=t.date_col.epoch_days())
+        ┏━━━━━━━━━━━━┳━━━━━━━┓
+        ┃ date_col   ┃ epoch ┃
+        ┡━━━━━━━━━━━━╇━━━━━━━┩
+        │ date       │ int64 │
+        ├────────────┼───────┤
+        │ 2020-01-01 │ 18262 │
+        └────────────┴───────┘
+        """
+        return ops.EpochDays(self).to_expr()
```

Comment:
Do we need a new op here? Is there a benefit to not defining this in terms of the existing `delta` method + a literal for the epoch? Doing it that way would avoid the need to define this op for every backend, instead relying on the more general `delta`.

Reply:
Oh, I see. I misinterpreted `delta` as not-the-method-but-the-concept, in which case, yeah maybe we can avoid the sqlglot munging and also a new op.